students/Brayden Borzyskowski/Lab1/Lab1Part4/Lab1Part4.ino

```diff
@@ -0,0 +1,176 @@
+unsigned long debounceDelay = 50;    // the debounce time; increase if the output flickers
+
+boolean lastButton = LOW; // Last Button State
+boolean currentButton = LOW; // Current Button State
+int ledMode = 0;            // Cycle between LED states
+
+void setup() 
+{
+pinMode (RLED, OUTPUT); // Set Red LED as Output
+pinMode (GLED, OUTPUT); // Set Green LED as Output
+pinMode (BLED, OUTPUT); // Set Blue LED as Output
+pinMode (BUTTON, INPUT); // Set button as input (not required)
+Serial.begin(9600);
+}
+ /*
+* Debouncing Function
+* Pass it the previous button state,
+* and get back the current debounced button state. 
+*/
+boolean debounce(boolean last)
+{
+  boolean current = digitalRead(BUTTON);  // Read the button state 
+  if (last != current)                    // If it's different...
+  {
+    delay(3);                             // Wait 5ms
+    current = digitalRead(BUTTON);        // Read it again
+  }
+  return current;                         // Return the current value
+}
+
+/*
+* LED Mode Selection
+* Pass a number for the LED state and set it accordingly
+* Note, since the RGB LED is COMMON ANODE, you must set the 
+* cathode pin for each color LOW for that color to turn ON. 
+*/
+void setMode(int mode)
+{
+//Blinking White
+  if (mode == 1)
+  { 
+    digitalWrite(RLED, HIGH); 
+    digitalWrite(GLED, HIGH); 
+    digitalWrite(BLED, HIGH);
+    delay(300);
+
+    digitalWrite(RLED, LOW); 
+    digitalWrite(GLED, LOW); 
+    digitalWrite(BLED, LOW);
+    delay(300);
+   
+
+  }
+  //Purple fading
+  else if (mode == 2) 
+  {
+    static unsigned long lastUpdate = 0;
+    unsigned long timeNow = millis(); 
+         // Fade from bright to dim
+         for (int brightness = 255; brightness >= 0; brightness--) 
+          {
+            analogWrite(RLED, redValue * brightness / 255);
+            analogWrite(BLED, blueValue * brightness / 255);
+            analogWrite(GLED, greenValue * brightness / 255);
+
+            
+          }
+
+         // Fade from dim to bright
+         for (int brightness = 0; brightness <= 255; brightness++) 
+          {
+            analogWrite(RLED, redValue * brightness / 255);
+            analogWrite(BLED, blueValue * brightness / 255);
+            analogWrite(GLED, greenValue * brightness / 255);
+            
+          }
+
+  }
+  //Start Red then Fade to Blue and back repeating
+  else if (mode == 3) 
+  {
+    analogWrite(GLED, 0);
+    static unsigned long lastUpdate = 0;
+    unsigned long timeNow = millis(); 
+    // Fade from red to blue
+    for (int brightness = 0; brightness <= 255; brightness++) 
+      {
+      analogWrite(RLED, 255 - brightness);
+      analogWrite(BLED, brightness);
+      }
+
+    // Fade from blue to red
+    for (int brightness = 255; brightness >= 0; brightness--) 
+      {
+      analogWrite(RLED, 255 - brightness);
+      analogWrite(BLED, brightness);
+      
+      }
+  }
+
+  else if (mode == 4)
+  {
+    digitalWrite(RLED, HIGH); 
```

Comment:
A challenge for the future: try to think if you can code something like this mode more efficiently/concisely, so you don't have blocks of repeated code.